### PR TITLE
S3 file lists - Improve handling edge cases

### PR DESCRIPTION
**Pull request by avaczi**
* try to find a better naming
* raise an error in the specific case of Access Denied (approximately matching 403)
* handle 'no file found' smoother

### CODE REVIEW

File: zmon_worker_monitor/builtins/plugins/s3.py [hunk 1]
```diff
@@ -156,13 +158,17 @@ class S3FileList(object):
...
         """
-        return [dict(zip(['file_name', 'last_modified', 'size'],
-                         [item['Key'], item['LastModified'], item['Size']]))
-                for item in self.__response['Contents']]
+        if self.__has_contents:
+            return [dict(zip(['file_name', 'last_modified', 'size'],
+                             [item['Key'], item['LastModified'], item['Size']]))
+                    for item in self.__response['Contents']]
+        else:
+            return {}
```
Comment (Jan-M) [inline zmon_worker_monitor/builtins/plugins/s3.py]: return []

Comment (avaczi) [inline zmon_worker_monitor/builtins/plugins/s3.py]: ouch.

**Comment by Jan-M** [comment]
:+1:

**Comment by mohabusama** [comment]
👍

**Comment by avaczi** [comment]
:+1:

**Pull request merged**